cmake/kokkos_enable_devices.cmake

```diff
@@ -42,12 +42,22 @@ IF(KOKKOS_ENABLE_OPENMP)
     #link omp library from LLVM lib dir
     get_filename_component(LLVM_BIN_DIR ${CMAKE_CXX_COMPILER_AR} DIRECTORY)
     COMPILER_SPECIFIC_LIBS(Clang "${LLVM_BIN_DIR}/../lib/libomp.lib")
+  ELSEIF(KOKKOS_CXX_COMPILER_ID STREQUAL NVIDIA)
+    SET(KOKKOS_CXX_COMPILER_ID_BACKUP ${KOKKOS_CXX_COMPILER_ID})
+    SET(KOKKOS_CXX_COMPILER_ID ${KOKKOS_CXX_HOST_COMPILER_ID})
+    COMPILER_SPECIFIC_FLAGS(
+      Clang      -Xcompiler ${ClangOpenMPFlag}
```

Comment:
Should we instead add an optional flag to the `COMPILER_SPECIFIC_FLAGS` macro that takes an optional compiler ID which defaults to KOKKOS_CXX_COMPILER_ID?

Reply:
I pushed a corresponding commit. I couldn't pass `${KOKKOS_CXX_HOST_COMPILER_ID}` directly, though, since that would conflict with parsing the compiler flags.